src/TestData/resources/js/forms.js

```diff
@@ -0,0 +1,111 @@
+import API, { CancelToken } from './api';
+// Common
+import {
+	processHasErrors,
+	updateDescription,
+	showRequestError,
+	updateProgerssBar,
+	generationStart,
+} from './utils';
+
+const { __, sprintf } = wp.i18n;
+const generateFormsBtn = document.querySelector( '#give-test-data-generate-forms' );
+
+const getData = () => {
+	return {
+		count: parseInt( document.querySelector( '#give-test-data-forms-count' ).value ),
+		template: getSelectedTemplate(),
+		setGoal: document.querySelector( '#give-test-data-forms-set-goal' ).checked,
+		setTC: document.querySelector( '#give-test-data-forms-set-tc' ).checked,
+	};
+};
+
+const getSelectedTemplate = () => {
+	const templates = document.querySelectorAll( 'input[name*="give_test_data_form_template"]:checked' );
+
+	if ( templates.length >= 2 ) {
+		return 'random';
+	}
+
+	return templates[ 0 ].value;
+};
+
+const generateForms = ( e ) => {
+	e.preventDefault();
+
+	const { count, template, setGoal, setTC } = getData();
+
+	// Check the donors count
+	if ( ! count ) {
+		// eslint-disable-next-line no-undef
+		return new Give.modal.GiveWarningAlert( {
+			modalContent: {
+				title: __( 'Enter number of forms', 'give-test-data' ),
+				desc: __( 'You must enter the number of forms to generate', 'give-test-data' ),
+				cancelBtnTitle: __( 'OK', 'give-test-data' ),
+			},
+		} ).render();
+	}
+
+	// eslint-disable-next-line no-undef
+	new Give.modal.GiveFormModal( {
```

Comment:
It might be worth extracting a small framework out of the logic that is shared across each page.

Reply:
I agree. The current implementation is very limited.